models/repo.go

```diff
@@ -1635,6 +1646,18 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 		}
 	}
 
+	projects, err := getProjects(sess, ProjectSearchOptions{
+		RepoID: repoID,
+	})
+	if err != nil {
+		return fmt.Errorf("get projects: %v", err)
+	}
+	for i := range projects {
+		if err := deleteProjectByID(sess, projects[i].ID); err != nil {
```

Comment:
It might be better to just delete the projects, projectissues and projectboards through the use of SQL commands that just delete them without instantiating them.